packages/core/admin/admin/src/pages/HomePage/index.js

```diff
@@ -8,13 +8,15 @@ import { FormattedMessage } from 'react-intl';
 import styled from 'styled-components';
 import { Helmet } from 'react-helmet';
 import { useHistory } from 'react-router-dom';
-import { LoadingIndicatorPage } from '@strapi/helper-plugin';
+import { LoadingIndicatorPage, useGuidedTour } from '@strapi/helper-plugin';
 import { Layout } from '@strapi/design-system/Layout';
 import { Main } from '@strapi/design-system/Main';
 import { Box } from '@strapi/design-system/Box';
 import { Grid, GridItem } from '@strapi/design-system/Grid';
 import Logo from '../../assets/images/homepage-logo.png';
 import { useModels } from '../../hooks';
+import isGuidedTourCompleted from '../../components/GuidedTour/utils/isGuidedTourCompleted';
+import GuidedTourHomepage from '../../components/GuidedTour/Homepage';
```

Comment:
Would it be possible to make this a code-splitting point, by using `React.lazy` and dynamic imports? This way it would not increase the bundlesize of the main bundle if the feature isn't needed (non admin-users, already completed ...).

Reply:
Not so sure about the advantages of splitting the modal I think that we should keep the splitting at the routes level for the moment.

Reply:
Fair point what is the difference in the build between having the tour and not?

If it makes a difference let's split otherwise let's keep it this way. WDYT?

Reply:
Sounds good to me. There should be no difference in the build (both chunks need to be build out), but only in case you are able to see the tour (as super admin) the chunk containing the code for the tour will be requested, but the user. It doesn't only save bandwidth but also execution time and so on ...

@ronronscelestes I leave the decision whether to split up to you. If you need any help on comparing numbers, let me know :)